packages/beeq/src/components/drawer/bq-drawer.tsx

```diff
@@ -0,0 +1,250 @@
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private handleFooterSlotChange = () => {
+    this.hasFooter = hasSlotContent(this.footerElem, 'footer');
+  };
+
+  private handleHide = async () => {
+    const ev = this.bqClose.emit(this.el);
+    if (!ev.defaultPrevented) {
+      await leave(this.drawerElem);
+      this.open = false;
+      this.handleTransitionEnd();
+    }
+  };
+
+  private handleShow = async () => {
+    const ev = this.bqOpen.emit(this.el);
+    if (!ev.defaultPrevented) {
+      this.open = true;
+      await enter(this.drawerElem);
+      this.handleTransitionEnd();
+    }
+  };
+
+  private handleTransitionEnd = () => {
+    if (this.open) {
+      this.bqAfterOpen.emit();
+      return;
+    }
+    this.bqAfterClose.emit();
+  };
+
+  private getMoveTranslate = (): string => {
+    switch (this.placement) {
+      case 'right':
+        return `translate-x-full`;
+      case 'left':
+        return `-translate-x-full`;
+      default:
+        return '';
+    }
+  };
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <Host
+        class={{ 'is-hidden': !this.open }}
+        aria-hidden={!this.open ? 'true' : 'false'}
+        hidden={!this.open ? 'true' : 'false'}
+        role="dialog"
+      >
+        <div class={{ 'bq-drawer-backdrop': this.enableBackdrop }}></div>
+        <div
+          class={{ [`bq-drawer ${this.placement}`]: true }}
+          data-transition-enter="transition-all ease-out duration-300"
+          data-transition-enter-start={`opacity-0 ${this.getMoveTranslate()}`}
+          data-transition-enter-end="opacity-100"
+          data-transition-leave="transition-all ease-in duration-300"
+          data-transition-leave-start="opacity-100"
+          data-transition-leave-end={`opacity-0 ${this.getMoveTranslate()}`}
+          ref={(div) => (this.drawerElem = div)}
+          part="wrapper"
+        >
+          <main class="bq-drawer__content" part="content">
+            <header class="bq-drawer__header" part="header">
+              <div class="bq-drawer__title" part="title">
+                <slot name="title" />
+              </div>
+              <div class="flex" role="button" part="button-close">
+                <slot name="button-close">
+                  <bq-button
+                    class="bq-drawer__close focus-visible:focus"
+                    appearance="text"
+                    size="small"
+                    slot="button-close"
+                    onClick={() => this.hide()}
+                  >
+                    <bq-icon weight="bold" name="x" role="img" title="Close" />
+                  </bq-button>
+                </slot>
+              </div>
+            </header>
+            <div class="bq-drawer__body" part="body">
+              <slot name="body" />
```

Comment:
I would keep this as default slot.
```suggestion
              <slot />
```